### PR TITLE
Fix test for Windows line endings

### LINKED ISSUE #86294: [CI] MultiCommandTests testErrorDisplayedWithDefault failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/fwkrw5dtfngdq/tests/:server:test/org.elasticsearch.cli.MultiCommandTests/testErrorDisplayedWithDefault

**Reproduction line:**
`gradlew ':server:test' --tests "org.elasticsearch.cli.MultiCommandTests.testErrorDisplayedWithDefault" -Dtests.seed=F9A116F9C72723D9 -Dtests.locale=th-TH-u-nu-thai-x-lvariant-TH -Dtests.timezone=Asia/Tehran -Druntime.java=18`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cli.MultiCommandTests&tests.test=testErrorDisplayedWithDefault

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: "ERROR: Dummy error\n"
     but: was "ERROR: Dummy error\r\n"

  at __randomizedtesting.SeedInfo.seed([F9A116F9C72723D9:9AC34FAE09D9AD87]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.cli.MultiCommandTests.testErrorDisplayedWithDefault(MultiCommandTests.java:224)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

### LINKED ISSUE #86341: [CI] FileTokensToolTests testListTokens failing
Same failure also happens in `FileTokensToolTests » testListTokensByPrincipal`. Line breaks are interpreted incorrectly.

**Build scan:**
https://gradle-enterprise.elastic.co/s/fy773g3kaycsq/tests/:x-pack:qa:security-tools-tests:test/org.elasticsearch.xpack.security.authc.service.FileTokensToolTests/testListTokens

**Reproduction line:**
`gradlew ':x-pack:qa:security-tools-tests:test' --tests "org.elasticsearch.xpack.security.authc.service.FileTokensToolTests.testListTokens" -Dtests.seed=232A7AD3A7EF5AB9 -Dtests.locale=es-VE -Dtests.timezone=Asia/Makassar -Druntime.java=18`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.authc.service.FileTokensToolTests&tests.test=testListTokens

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a string containing "elastic/fleet-server/server_1\nelastic/fleet-server/server_2\nelastic/fleet-server/server_3"
     but: was "elastic/fleet-server/server_1
elastic/fleet-server/server_2
elastic/fleet-server/server_3
"

  at __randomizedtesting.SeedInfo.seed([232A7AD3A7EF5AB9:1B8178450218C40E]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.authc.service.FileTokensToolTests.testListTokens(FileTokensToolTests.java:266)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by williamrandolph**
Adjustments to the MockTerminal class in #86198 and elsewhere have broken tests that depended on specific line endings. In many cases, this was not what the test was actually trying to check for. This fix makes the strings as system-agnostic as possible, in several cases using the `String#lines` method to handle line endings in a system-agnostic way.

Closes #86294. Closes #86341

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cli/MultiCommandTests.java [hunk 1]
```diff
@@ -221,7 +221,7 @@ public void testErrorDisplayedWithDefault() throws Exception {
         multiCommand.subcommands.put("throw", new ErrorThrowingSubCommand());
         executeMain("throw", "--silent");
         assertThat(terminal.getOutput(), is(emptyString()));
-        assertThat(terminal.getErrorOutput(), equalTo("ERROR: Dummy error\n"));
+        assertThat(terminal.getErrorOutput(), equalTo("ERROR: Dummy error" + System.lineSeparator()));
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/cli/MultiCommandTests.java:224]: We could probably just remove the line ending altogether and use startsWith? This test is trying to assert errors are printed when --silent is used, not check anything about terminal output.

**Comment by pugnascotia** [comment]
I pushed some more fixes to this branch since the problem is causing a lot of build failures.

File: server/src/test/java/org/elasticsearch/cli/MultiCommandTests.java (hunk 1 above)
Comment (williamrandolph) [reply server/src/test/java/org/elasticsearch/cli/MultiCommandTests.java]: Good call. Thanks!

File: distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ListKeyStoreCommandTests.java [hunk 2]
```diff
@@ -41,23 +45,23 @@ public void testEmpty() throws Exception {
         createKeystore(password);
         terminal.addSecretInput(password);
         execute();
-        assertEquals("keystore.seed\n", terminal.getOutput());
+        assertThat(terminal.getOutput().lines().toList(), contains("keystore.seed"));
```
Comment (rjernst) [inline distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ListKeyStoreCommandTests.java:48]: No need to parse it using `lines()`, just use `containsString`?

File: distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ShowKeyStoreCommandTests.java [hunk 3]
```diff
@@ -62,7 +63,7 @@ public void testPrintSingleValueToTerminal() throws Exception {
         createKeystore(password, "reindex.ssl.keystore.password", value);
         terminal.addSecretInput(password);
         execute("reindex.ssl.keystore.password");
-        assertEquals(value + "\n", terminal.getOutput());
+        assertThat(terminal.getOutput(), equalTo(value + System.lineSeparator()));
```
Comment (rjernst) [inline distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ShowKeyStoreCommandTests.java:66]: Can we not depend on the line separator here?

File: x-pack/qa/security-tools-tests/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetPasswordToolTests.java [hunk 4]
```diff
@@ -292,8 +292,7 @@ public void testAutoInteractiveModesMutuallyExclusive() throws Exception {
 
     public void testAutoBatchSilent() throws Exception {
         execute(randomFrom("--silent", "-s"), randomFrom("--batch", "-b"), randomFrom("-u", "--username"), user);
-        String output = terminal.getOutput();
-        assertThat(output, hasLength(21)); // password + new line char
```
Comment (rjernst) [inline x-pack/qa/security-tools-tests/src/test/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetPasswordToolTests.java:296]: This seems like a place where we do actually care about the newline. Using `lines()` would mask if we were not printing any newline at all. Could we still assert the length?

**Review by rjernst — APPROVED** [review]
Looks good, I have a few suggestions.

File: distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ListKeyStoreCommandTests.java (hunk 2 above)
Comment (williamrandolph) [reply distribution/tools/keystore-cli/src/test/java/org/elasticsearch/cli/keystore/ListKeyStoreCommandTests.java:48]: My thinking here was that `lines()` gives us a list containing a string which must be exactly `keystore.seed`, which is a stronger check than whether whether the output contains a substring.

**Comment by williamrandolph** [comment]
@elasticmachine please run elasticsearch-ci/part-1-windows

**Comment by williamrandolph** [comment]
Since the failure here is unrelated, I'm going to go ahead and merge. I'll follow up tomorrow with muting if anything isn't working as planned.

**Pull request merged**